### PR TITLE
Correcting DELIVER_USER env var name

### PR DESCRIPTION
**Pull request by dombarnes**
Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes:
- [X] Run `rspec` for all tools you modified
- [X] Run `rubocop -a` to ensure the code style is valid
- [X] We currently don't accept new actions, please publish a plugin instead, more information in [Plugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md)

Before submitting a pull request, we appreciate if you create an issue first to discuss the change :+1:

Fix for incorrect ENV VAR in `deliver` options as discussed in https://github.com/fastlane/fastlane/issues/6478

### CODE REVIEW

File: deliver/lib/deliver/options.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ def self.available_options
       [
         FastlaneCore::ConfigItem.new(key: :username,
                                      short_option: "-u",
-                                     env_name: "DELIVER_USERNAME",
+                                     env_name: "DELIVER_USER",
```
Comment (KrauseFx) [inline deliver/lib/deliver/options.rb]: This is a breaking change, we'd have to support both parameter. The easiest way to do so is to change the default value to

```
default_value: user || ENV["DELIVER_USER"]
```

**Review by KrauseFx — CHANGES_REQUESTED** [review]

File: deliver/lib/deliver/app_screenshot.rb [hunk 2]
```diff
@@ -137,7 +137,7 @@ def self.devices
     def self.calculate_screen_size(path)
       size = FastImage.size(path)
 
-      UI.user_error!("Could not find or parse file at path '#{path}'") if size.nil? or size.count == 0
+      UI.user_error!("Could not find or parse file at path '#{path}'") if size.nil? or size.count.zero?
```
Comment (KrauseFx) [inline deliver/lib/deliver/app_screenshot.rb]: These are actually not changes we want in here. We recently updated the rubocop rules to not include this

**Review by KrauseFx — CHANGES_REQUESTED** [review]

**Comment by dombarnes** [comment]
Thanks @KrauseFx. I've just revered those changes and added the change to the default so as not to break backwards. Upon running `rake` I get one test error though, which seems to be failing due to reading an ENV var on my system. Not sure if this test needs adjusting at all?

````
1) Fastlane::CrashlyticsBetaUserEmailFetcher Email fetching does not find an email
Failure/Error: expect(fetcher.fetch).to be_nil
expected: nil
     got: "user@example.com"# fastlane/spec/setup/crashlytics_beta_user_email_fetcher_spec.rb:22:in `block (3 levels) in <top (required)>'

Failed examples:
fastlane/spec/setup/crashlytics_beta_user_email_fetcher_spec.rb:20 # Fastlane::CrashlyticsBetaUserEmailFetcher Email fetching does not find an email

================
| Exceptions 💣 |
================

fastlane: Command failed with status (1): [bundle exec rspec --format documentation -...]
spaceship: Command failed with status (7): [bundle exec rspec --format documentation -...]
rake aborted!
All tests did not complete successfully. Search for '[[FAILURE]]' in the build logs.
/Users/domster83/projects/fastlane/rakelib/test_all.rake:152:in `block in <top (required)>'
Tasks: TOP => default => test_all
(See full trace by running task with --trace)
````

File: deliver/lib/deliver/detect_values.rb [hunk 3]
```diff
@@ -22,7 +22,7 @@ def find_app_identifier(options)
 
     def find_app(options)
       search_by = options[:app_identifier]
-      search_by = options[:app] if search_by.to_s.length == 0
+      search_by = options[:app] if search_by.to_s.length.zero?
```
Comment (ohwutup) [inline deliver/lib/deliver/detect_values.rb]: I think we would like to keep this as `== 0`

File: deliver/lib/deliver/upload_screenshots.rb [hunk 4]
```diff
@@ -82,7 +82,7 @@ def collect_screenshots_for_languages(path)
         end
 
         files = Dir.glob(File.join(lng_folder, "*.#{extensions}"), File::FNM_CASEFOLD).sort
-        next if files.count == 0
+        next if files.count.zero?
```
Comment (ohwutup) [inline deliver/lib/deliver/upload_screenshots.rb]: I think we would like to keep this as `== 0`

**Review by ohayon — CHANGES_REQUESTED** [review]
Hey @dombarnes, I just left a couple comments about keeping the 0 checks as `==` rather than `zero?`. Also, it looks like there are some merge conflicts, would you mind resolving those and we can move forward with the PR?

**Comment by asfalcone** [comment]
Closing this in favor of #7806

**Pull request closed without merging**